internal/telemetry/middleware.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021-2022 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"context"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	telemetryv1 "github.com/cerbos/cerbos/api/genpb/cerbos/telemetry/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+const collectorBufferSize = 64
+
+var totalReqCount uint64
```

Comment:
Would it make sense for this to be a field on the `statsInterceptors` struct similar to the tallies?